src/components/ButtonGroupPicker/index.js

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import withReduxForm from './../../libs/hocs/withReduxForm';
+import { uniqueId } from '../../libs/utils';
+import RenderIf from '../RenderIf';
+import { Provider } from './context';
+import {
+    StyledButtonGroup,
+    StyledContainer,
+    StyledErrorText,
+    StyledHelpText,
+    StyledLegend,
+} from './styled';
+import RequiredAsterisk from '../RequiredAsterisk';
+
+/**
+ * ButtonGroupPicker can be used to group related options. The ButtonGroupPicker will control the selected state of its child ButtonOption.
+ */
+class ButtonGroupPicker extends Component {
+    constructor(props) {
+        super(props);
+        this.groupNameId = this.props.name || uniqueId('options');
+        this.errorMessageId = uniqueId('error-message');
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    getErrorMessageId() {
+        const { error } = this.props;
+        if (error) {
+            return this.errorMessageId;
+        }
+        return undefined;
+    }
+
+    handleOnChange(event) {
+        const { value: eventValue, checked } = event.target;
+        const { value, multiple, onChange } = this.props;
+
+        if (!multiple) {
+            return onChange(eventValue);
+        }
+
+        if (checked && Array.isArray(value)) {
+            return onChange(value.concat([eventValue]));
+        }
+        if (checked && !Array.isArray(value)) {
+            return onChange(eventValue);
```

Comment:
What happens when the user does not pass an initial value and then it is undefined or if it passes a wrong value initially?
I think here we should do the following:
```suggestion
            return onChange([eventValue]);
```
In this way, we can recover from the initial undefined or wrong value and the component will work as expected.